src/Action/Server.hs

```diff
@@ -60,11 +63,12 @@ actionServer cmd@Server{..} = do
     log <- logCreate (if logs == "" then Left stdout else Right logs) $
         \x -> BS.pack "hoogle=" `BS.isInfixOf` x && not (BS.pack "is:ping" `BS.isInfixOf` x)
     putStrLn . showDuration =<< time
-    evaluate spawned
+    _ <- evaluate spawned
     dataDir <- maybe getDataDir pure datadir
-    haddock <- maybe (pure Nothing) (fmap Just . canonicalizePath) haddock
+    haddock' <- maybe (pure Nothing) (fmap Just . canonicalizePath) haddock
```

Comment:
I actually thing this is more dangerous than before. Now we have satisfied this warning if you accidentally type `haddock` on the next line it works and goes wrong. When we shadowed the variable, we eliminated that problem by construction.